packages/compass-assistant/src/compass-assistant-provider.tsx

```diff
@@ -41,6 +41,8 @@ export type AssistantMessage = UIMessage & {
      *  Used for warning messages in cases like using non-genuine MongoDB.
      */
     isPermanent?: boolean;
+    /** The source of the message (i.e. the entry point used) */
+    source?: 'explain plan' | 'performance insights' | 'connection error';
```

Comment:
I'm wondering if we should treat normal typed prompts as an "entry point" and then this becomes required and we can easily segment all messages. Like I'm worried about the "ergonomics" of filtering by no source and the ambiguity introduced by a field that's not there. Maybe a question for the analytics team?